source/logrepl/combined_iterator.go

```diff
@@ -0,0 +1,59 @@
+package logrepl
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jackc/pgx/v4"
+)
+
+const (
+	TypeSnapshot int = iota
+	TypeCDC
+)
+
+// CombinedIterator binds a CDCIterator and a SnapshotIterator together for
+// coordinating the hand-off from snapshot to CDC operation modes.
+type CombinedIterator struct {
+	cdc  *CDCIterator
+	snap *SnapshotIterator
+}
+
+func NewCombinedIterator(ctx context.Context, conn *pgx.Conn, cfg Config) (*CombinedIterator, error) {
```

Comment:
We should probably rename the current `Config` struct to `CDCConfig` and create a separate `Config` struct for this iterator.